### PR TITLE
Added config setting JobSettings.CleanupJobsThresholdDays

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for https://github.com/mattermost/mattermost-server/pull/18646
- Added ``config.json`` only config setting JobSettings.CleanupJobsThresholdDays

### CODE REVIEW

**Comment by cwarnermm** [comment]
@agnivade - Is a Telemetry update required as well for this new configuration setting?

File: source/configure/configuration-settings.rst [hunk 1]
```diff
@@ -12733,9 +12733,25 @@ This setting isn't available in the System Console and can only be set in ``conf
...
+~~~~~~~~~~~~~~~~~
+
+Clean Up Old Database Jobs
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+|all-plans| |self-hosted|
+
+This setting isn't available in the System Console and can only be set in ``config.json``.
+
+Defines the threshold in hours beyond which older completed database jobs are removed. This setting applies to both MySQL and PostgreSQL databases, is disabled by default, and must be set to ``0`` to be enabled.
```
Comment (agnivade) [inline source/configure/configuration-settings.rst:12750]: It must be set to something greater than or equal to 0 to be enabled.

**Comment by agnivade** [comment]
Yes, good point. A telemetry update is needed. I'll send a PR for that.

File: source/configure/configuration-settings.rst (hunk 1 above)
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:12750]:
```suggestion
Defines the threshold in hours beyond which older completed database jobs are removed. This setting applies to both MySQL and PostgreSQL databases, is disabled by default, and must be set to a value greater than or equal to ``0`` to be enabled.
```

**Review by justinegeffen — APPROVED** [review]

**Review by agnivade — APPROVED** [review]

**Pull request merged**